tcs-persistence/src/main/resources/queries/traineeInfoForConnection.sql

```diff
@@ -31,8 +32,12 @@ from (
           cm.programmeMembershipUuid,
           max(cm.curriculumEndDate) curriculumEndDate
       from CurriculumMembership cm
+      WHERECLAUSE1 -- where personId in ()
       group by cm.programmeMembershipUuid
   ) latestCm on latestCm.programmeMembershipUuid = pm1.uuid
   left join Programme prg on prg.id = pm1.programmeId
+  WHERECLAUSE2 -- where p.id in ()
```

Comment:
Looks good & comments are really useful.

I'd check this before committing but: I think you could replace `WHERECLAUSE2` by modifying traineeInfoForConnection.sql#L4 with `   p.id personId`.  There may be something to change in the Java to match the alias but if it improves the readability without hurting performance then it's a positive difference.